backend/src/main/java/ai/verta/modeldb/versioning/BlobDAORdbImpl.java

```diff
@@ -479,28 +529,77 @@ boolean childContains(Set<?> list, Set<?> sublist) {
 
   @Override
   public MergeRepositoryCommitsRequest.Response mergeCommit(
-      RepositoryFunction repositoryFunction, MergeRepositoryCommitsRequest request)
+      RepositoryDAO repositoryDAO, MergeRepositoryCommitsRequest request)
       throws ModelDBException, NoSuchAlgorithmException {
+
+    boolean invalidParam = false;
+    if (!request.getBranchA().isEmpty()
+        && !request.getBranchB().isEmpty()
+        && !request.getCommitShaA().isEmpty()
+        && !request.getCommitShaB().isEmpty()) {
+      invalidParam = true;
+    } else if (!request.getBranchA().isEmpty() && !request.getCommitShaA().isEmpty()) {
+      invalidParam = true;
+    } else if (request.getBranchA().isEmpty() && request.getCommitShaA().isEmpty()) {
+      invalidParam = true;
+    } else if (!request.getBranchB().isEmpty() && !request.getCommitShaB().isEmpty()) {
+      invalidParam = true;
+    } else if (request.getBranchB().isEmpty() && request.getCommitShaB().isEmpty()) {
+      invalidParam = true;
+    }
```

Comment:
if it's the same code as in computeRepositoryDiff (except request type) consider reusing the code

Reply:
fixed